hadoop-hdds/rocks-native/src/main/patches/rocks-native.patch

```diff
@@ -16,592 +16,557 @@
++};
++
++}  // namespace ROCKSDB_NAMESPACE
++
++#endif  // ROCKSDB_LITE
+diff --git a/src.mk b/src.mk
+index b94bc43ca..95bfff274 100644
+--- a/src.mk
++++ b/src.mk
+@@ -343,6 +343,8 @@ TOOL_LIB_SOURCES =                                              \
+   tools/ldb_tool.cc                                             \
+   tools/sst_dump_tool.cc                                        \
+   utilities/blob_db/blob_dump_tool.cc                           \
++  tools/raw_sst_file_reader.cc                                  \
++  tools/raw_sst_file_reader_iterator.cc                         \
+
+ ANALYZER_LIB_SOURCES =                                          \
+   tools/block_cache_analyzer/block_cache_trace_analyzer.cc      \
+diff --git a/tools/raw_sst_file_reader.cc b/tools/raw_sst_file_reader.cc
+new file mode 100644
+index 000000000..1693bd1e6
+--- /dev/null
++++ b/tools/raw_sst_file_reader.cc
+@@ -0,0 +1,285 @@
++//  Copyright (c) 2011-present, Facebook, Inc.  All rights reserved.
++//  This source code is licensed under both the GPLv2 (found in the
++//  COPYING file in the root directory) and Apache 2.0 License
++//  (found in the LICENSE.Apache file in the root directory).
++//
++#ifndef ROCKSDB_LITE
++
++#include "rocksdb/raw_sst_file_reader.h"
++
++#include <chrono>
++#include <cinttypes>
++#include <iostream>
++#include <map>
++#include <memory>
++#include <sstream>
++#include <vector>
++
++#include "db/blob/blob_index.h"
++#include "db/memtable.h"
++#include "db/write_batch_internal.h"
++#include "options/cf_options.h"
++#include "port/port.h"
++#include "rocksdb/db.h"
++#include "rocksdb/env.h"
++#include "rocksdb/iterator.h"
++#include "rocksdb/slice_transform.h"
++#include "rocksdb/status.h"
++#include "rocksdb/table_properties.h"
++#include "rocksdb/utilities/ldb_cmd.h"
++#include "table/block_based/block.h"
++#include "table/block_based/block_based_table_builder.h"
++#include "table/block_based/block_based_table_factory.h"
++#include "table/block_based/block_builder.h"
++#include "table/format.h"
++#include "table/meta_blocks.h"
++#include "table/plain/plain_table_factory.h"
++#include "table/table_reader.h"
++#include "tools/raw_sst_file_reader_iterator.h"
++#include "util/compression.h"
++#include "util/random.h"
++#include "db/dbformat.h"
++#include "file/writable_file_writer.h"
++#include "options/cf_options.h"
++
++namespace ROCKSDB_NAMESPACE {
++
++struct RawSstFileReader::Rep {
++  Options options;
++  EnvOptions soptions_;
++  ReadOptions read_options_;
++  ImmutableOptions ioptions_;
++  MutableCFOptions moptions_;
++  InternalKeyComparator internal_comparator_;
++  std::unique_ptr<TableProperties> table_properties_;
++  std::unique_ptr<TableReader> table_reader_;
++  std::unique_ptr<RandomAccessFileReader> file_;
++
++  Rep(const Options& opts, bool verify_checksum, size_t readahead_size)
++      : options(opts),
++        soptions_(EnvOptions()),
++        read_options_(verify_checksum, false),
```

Comment:
You can providing an options to verify checksum but it is never used as it is in [sst_file_dumper](https://github.com/facebook/rocksdb/blob/main/table/sst_file_dumper.cc#L202-L207). Is my understanding correct, if so, we should add that or remove the option.

Reply:
We have that option here but we are always setting it to be true.

Reply:
I'm inclined towards [You aren't gonna need it](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it) principle prefer to do things when needed.

Having said that, you can leave it as it is.